core/store/src/trie/mod.rs

```diff
@@ -757,6 +769,101 @@ impl Trie {
     pub fn get_trie_nodes_count(&self) -> TrieNodesCount {
         self.storage.get_trie_nodes_count()
     }
+
+    pub fn get_proof(
+        &self,
+        root: &CryptoHash,
+        key: &[u8],
+    ) -> Result<Vec<TrieMerkleProofItem>, StorageError> {
```

Comment:
@akhi3030 suggested empty vec instead of `Option<Vec<>>`. I agree that probably it's better that way. Any thoughts @akhi3030 ?

Reply:
My thinking was what is the difference between `None` and an empty vec.  If they both represent the same value, then we should not have two ways to represent a single value.  However, if they actually do represent different values, then we should absolutely have an option here.

Reply:
I added some docs @mina86 . Still unsure of what to do with regards to adding `Option` or keeping it the way it is.

Reply:
Note that by ‘key was not found’ I also mean that there was some kind of error.  Since the proof generation happened after we’ve retrieved the value for the key, this should be reported as error.  Or better still, we might wanna look into doing the proof while doing the lookup.